### DISCUSSION TITLE
How to hide a flyout automatically after a certain time interval?

### DISCUSSION THREAD

**Discussion by klondikedragon** [bdlukaa/fluent_ui#741] [Q&A] [answered]
After #714 it's not as obvious to me how to close a flyout (if it's still open) automatically after a period of time. Before, the flyout controller could be used without the build context to simply call the close method on it.

Now, it seems that the way to close the flyout requires `Flyout.of(context).close()` - it feels unsafe to put code that might trigger this inside of future that gets created in the build method (where the context is needed). What's the right way to do something like this now?

```dart
            flyoutController.showFlyout(
              barrierDismissible: false,
              builder: (context) {
                Future.delayed(const Duration(milliseconds: 1200)).then((_) {
                  if (flyoutController.isAttached &&
                      flyoutController.isOpen) {
                    // ignore: use_build_context_synchronously
                    Flyout.of(context).close();
                  }
                });
                return FlyoutContent(
                  child: Text("hello world"),
                );
              },
            );
```

**Comment by bdlukaa** [comment, marked as answer]
you can use `context.mounted` to ensure the flyout is on the screen

**Reply by klondikedragon** [reply]
Thanks! So far this is working and avoids an exception. I've found a different case where it seems to be a bug, I'll open an issue about it.
